Sources/NIOHTTP2/HTTP2StreamChannel.swift

```diff
@@ -998,3 +1007,9 @@ extension HTTP2StreamChannel {
         SynchronousOptions(channel: self)
     }
 }
+
+/// Events that can be sent by the application to be handled by the `HTTP2StreamChannel`
+public enum HTTP2StreamChannelEvent {
```

Comment:
So I think we'll probably end up regressing making this a public enum, as we can't easily evolve that as an API. I think we should store the "real" value in an enum, but have this wrapped in a struct that has a few `static func` constructors..

We'll also need tests for this.